netx/httptransport/http3transport_test.go

```diff
@@ -32,7 +39,7 @@ func TestUnitHTTP3TransportSuccess(t *testing.T) {
 }
 
 func TestUnitHTTP3TransportFailure(t *testing.T) {
-	txp := httptransport.NewHTTP3Transport(selfcensor.SystemDialer{}, dialer.TLSDialer{})
+	txp := httptransport.NewHTTP3Transport(httptransport.Config{selfcensor.SystemDialer{}, MockHTTP3Dialer{}, dialer.TLSDialer{}})
```

Comment:
Could you explicitly initialise fields? I think this makes the code more obvious.